moveit_core/robot_model/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.5)
 set(MOVEIT_LIB_NAME moveit_robot_model)
 
-add_library(${MOVEIT_LIB_NAME}
+add_library(${MOVEIT_LIB_NAME} SHARED
```

Comment:
out of curiosity, why is that suddenly needed? From what I see, It shouldn't be.

Reply:
It helps with compiling .so libraries. Apparently, libraries are static by default now (don't ask where this is documented)

Reply:
whether they default static or shared depends on `BUILD_SHARED_LIBS`, see the [cmake documentation](https://cmake.org/cmake/help/v3.16/command/add_library.html).

It would surely be some work, but I believe it should be possible to build a STATIC workspace too; I'm convinced no one ever looked into it though...
Static MoveIt libraries could actually be appreciated for some industry applications, so I'm not happy to see (1) more text added to the buildsystem (2) that might break or forbid corner-case build setups.

Reply:
I'm aware of `BUILD_SHARED_LIBS` and from what I've seen, shared libraries are supposed to be the default (but they are not). Most ros2 packages I've seen use `SHARED` to enforce shared libraries. Of course static workspaces should work as well and your points are valid, but I don't see why we should introduce this here and now. This PR blocks at least 3 engineers from progressing on other packages and revising the build system for corner cases is something we should consider once we actually have a working setup.

Reply:
I never said it should block, I was asking out of curiosity. It would make sense to me to add a comment with a TODO to investigate things like this (I'm sure it's not the only questionable detail in recent migration patches) somewhere or file issues upon merge.

Still, "we need this merged now, because of money" is a bad argument in open source, and you should be very well aware of it. It's how you ended up in this whole mess to begin with.

Reply:
I'll address it in the next weekly sync and note it as an issue. If you are free to investigate things like this you are very welcome ;)